src/cast/safe/trim.rs

```diff
@@ -0,0 +1,109 @@
+/// Cast from a larger number to a smaller one without changing sign.
+///
+/// The `TrimFrom<T>` trait is similar to `std::convert::From<T>`. However,
+/// while `std::convert::From<T>` performs a logical conversion, `TrimFrom<T>`
+/// performs a bitwise cast from a number to a number of the same sign but
+/// of a possibly smaller size. `TrimFrom<T>` will **never** increase the size
+/// of a number or change from an integer of one signedness to the other.
```

Comment:
Again, the value of unknown signedness is not clear to me.

And once you've hedged that this is only *possibly* a smaller size, how is this better than `AsPrimitive`?